sources/Interop/Windows/winrt/wrl/client/ComPtr`1.Manual.cs

```diff
@@ -4,150 +4,401 @@
+        /// </summary>
+        /// <param name="other">The target reference to copy the address of the current COM object to.</param>
+        /// <returns>The result of <see cref="IUnknown.QueryInterface"/> for the target type <typeparamref name="U"/>.</returns>
+        public readonly int CopyTo<U>(ref ComPtr<U> other)
+            where U : unmanaged
+        {
+            fixed (ComPtr<U>* p = &other)
+            {
+                return ((IUnknown*)ptr_)->QueryInterface(__uuidof<U>(), (void**)p->ReleaseAndGetAddressOf());
+            }
+        }
+
+        /// <summary>
+        /// Converts the current object reference to a type indicated by the given IID and assigns that to a target address.
+        /// </summary>
+        /// <param name="riid">The IID indicating the interface type to convert the COM object reference to.</param>
+        /// <param name="ptr">The target raw pointer to copy the address of the current COM object to.</param>
+        /// <returns>The result of <see cref="IUnknown.QueryInterface"/> for the target IID.</returns>
+        public readonly int CopyTo(Guid* riid, void** ptr)
+        {
+            return ((IUnknown*)ptr_)->QueryInterface(riid, ptr);
+        }
+
+        /// <summary>
+        /// Converts the current object reference to a type indicated by the given IID and assigns that to a target <see cref="ComPtr{T}"/> value.
+        /// </summary>
+        /// <param name="riid">The IID indicating the interface type to convert the COM object reference to.</param>
+        /// <param name="p">The target raw pointer to copy the address of the current COM object to.</param>
+        /// <returns>The result of <see cref="IUnknown.QueryInterface"/> for the target IID.</returns>
+        public readonly int CopyTo(Guid* riid, ComPtr<IUnknown>* p)
+        {
+            return ((IUnknown*)ptr_)->QueryInterface(riid, (void**)p->ReleaseAndGetAddressOf());
         }
 
+        /// <summary>
+        /// Converts the current object reference to a type indicated by the given IID and assigns that to a target <see cref="ComPtr{T}"/> value.
+        /// </summary>
+        /// <param name="riid">The IID indicating the interface type to convert the COM object reference to.</param>
+        /// <param name="other">The target reference to copy the address of the current COM object to.</param>
+        /// <returns>The result of <see cref="IUnknown.QueryInterface"/> for the target IID.</returns>
+        public readonly int CopyTo(Guid* riid, ref ComPtr<IUnknown> other)
+        {
+            fixed (ComPtr<IUnknown>* p = &other)
+            {
+                return ((IUnknown*)ptr_)->QueryInterface(riid, (void**)p->ReleaseAndGetAddressOf());
+            }
+        }
+
+        /// <inheritdoc/>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void Dispose()
         {
-            InternalRelease();
+            T* pointer = ptr_;
```

Comment:
Why was this changed from using `InternalRelease()`?

Reply:
Here we don't return the ref count so thought doing this manually and simplified could help the JIT with inlining?